From d51cc414b84db789a42a53c110a5b42819c5f3e4 Mon Sep 17 00:00:00 2001
From: Mengkejiergeli Ba <mengkejiergeli.ba@intel.com>
Date: Fri, 18 Mar 2022 18:07:50 +0800
Subject: [PATCH 09/10] msdkenc: Add more supported formats for DMABuf caps

More formats with DMABuf sink caps can be supported, that is:
VUYA, YUY2, Y210 and Y410 for msdkh265enc, VUYA and Y410 for msdkvp9enc.

With reference to iHD doc:
https://github.com/intel/media-driver/blob/master/docs/media_features.md#supported-encoding-input-format-and-max-resolution
---
 subprojects/gst-plugins-bad/sys/msdk/gstmsdkh265enc.c | 2 +-
 subprojects/gst-plugins-bad/sys/msdk/gstmsdkvp9enc.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkh265enc.c b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkh265enc.c
index 5fca672aab..e4b97d3b65 100644
--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkh265enc.c
+++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkh265enc.c
@@ -147,7 +147,7 @@ static GstStaticPadTemplate sink_factory = GST_STATIC_PAD_TEMPLATE ("sink",
     GST_PAD_SINK,
     GST_PAD_ALWAYS,
     GST_STATIC_CAPS (GST_MSDK_CAPS_STR (COMMON_FORMAT,
-            "{ NV12, P010_10LE }") "; "
+            "{ NV12, YUY2, VUYA, P010_10LE, Y210, Y410 }") "; "
         GST_MSDK_CAPS_MAKE_WITH_VA_FEATURE ("NV12")));
 #endif
 
diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvp9enc.c b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvp9enc.c
index da35769a68..fda0ccbc51 100644
--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvp9enc.c
+++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvp9enc.c
@@ -78,7 +78,7 @@ static GstStaticPadTemplate sink_factory = GST_STATIC_PAD_TEMPLATE ("sink",
     GST_PAD_SINK,
     GST_PAD_ALWAYS,
     GST_STATIC_CAPS (GST_MSDK_CAPS_STR (COMMON_FORMAT,
-            "{ NV12, P010_10LE }") "; "
+            "{ NV12, VUYA, P010_10LE, Y410 }") "; "
         GST_MSDK_CAPS_MAKE_WITH_VA_FEATURE ("NV12")));
 #endif
 
-- 
2.25.1

